### PR TITLE
Change of periodic-gcs-logs-cleanup to use googlecloudstorageclient instead of gsutil

### PR DESCRIPTION
**Pull request by Rajalakshmi-Girish**
Changing the existing periodic job `periodic-gcs-logs-cleanup` to use googlecloudstorageclient instead of gsutil to delete the objects older than the specified no. of days in the config file.
Removing the current logic of moving the logs to IBM COS s3 storage as per the below comment:
https://github.ibm.com/powercloud/container-dev/issues/1375#issuecomment-27545024

### CODE REVIEW

**Comment by ltccci** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/ppc64le-cloud/test-infra/pull/131#" title="Author self-approved">Rajalakshmi-Girish</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **mkumatag** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @mkumatag` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=ppc64le-cloud%2Ftest-infra).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/ppc64le-cloud/test-infra/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: config/jobs/periodic/housekeeping/main.go
Comment (mkumatag) [inline config/jobs/periodic/housekeeping/main.go:1]: can we have a separate project altogether for this task for better maintainability?

**Review by mkumatag — CHANGES_REQUESTED** [review]

File: config/jobs/periodic/housekeeping/main.go
Comment (mkumatag) [reply config/jobs/periodic/housekeeping/main.go:1]: https://github.com/ppc64le-cloud/gcs-cleaner.git here is the repository created.

File: config/jobs/periodic/housekeeping/cleanup-gcs-logs-periodics.yaml
Comment (mkumatag) [inline config/jobs/periodic/housekeeping/cleanup-gcs-logs-periodics.yaml:34]: do we need to specify any flags like gcs cred?

**Review by mkumatag — CHANGES_REQUESTED** [review]

File: config/jobs/periodic/housekeeping/cleanup-gcs-logs-periodics.yaml
Comment (Rajalakshmi-Girish) [reply config/jobs/periodic/housekeeping/cleanup-gcs-logs-periodics.yaml:34]: We have the default value for flag `gcs-cred-path` as `/etc/gcs-cred/service-account.json`.
For this test-pod, we are mounting the secret `gcs-credentials` to path `/etc/gcs-cred`, we don't need to pass it explicitly. Please see the below line:
https://github.com/ppc64le-cloud/gcs-cleaner/blob/main/main.go#L47

Comment (mkumatag) [reply config/jobs/periodic/housekeeping/cleanup-gcs-logs-periodics.yaml:34]: cool

**Review by mkumatag — APPROVED** [review]
/lgtm
/approve

**Comment by ltccci** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/ppc64le-cloud/test-infra/pull/131#pullrequestreview-599757645" title="Approved">mkumatag</a>*, *<a href="https://github.com/ppc64le-cloud/test-infra/pull/131#" title="Author self-approved">Rajalakshmi-Girish</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=ppc64le-cloud%2Ftest-infra).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/ppc64le-cloud/test-infra/blob/master/OWNERS)~~ [mkumatag]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by ltccci** [comment]
@Rajalakshmi-Girish: Updated the `job-config` configmap in namespace `prow` at cluster `default` using the following files:
 - key `cleanup-gcs-logs-periodics.yaml` using file `config/jobs/periodic/housekeeping/cleanup-gcs-logs-periodics.yaml`

<details>

In response to [this](https://github.com/ppc64le-cloud/test-infra/pull/131):

>Changing the existing periodic job `periodic-gcs-logs-cleanup` to use googlecloudstorageclient instead of gsutil to delete the objects older than the specified no. of days in the config file.
>Removing the current logic of moving the logs to IBM COS s3 storage as per the below comment:
>https://github.ibm.com/powercloud/container-dev/issues/1375#issuecomment-27545024


Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Pull request merged**
